desktop/core/src/desktop/js/apps/newimporter/utils/utils.test.ts

```diff
@@ -35,70 +82,228 @@ describe('convertToAntdColumns', () => {
+      { importerDataKey: 'importer-row__2', name: 'Charlie', age: null, city: null }
     ];
 
-    expect(convertToDataSource(columns, apiResponse)).toEqual(expectedOutput);
+    expect(convertToDataSource(apiResponse)).toEqual(expectedOutput);
+  });
+
+  it('should handle empty arrays in apiResponse', () => {
+    const apiResponse: FilePreviewTableData = {
+      name: [],
+      age: []
+    };
+
+    expect(convertToDataSource(apiResponse)).toEqual([]);
   });
 });
 
 describe('getDefaultTableName', () => {
-  it('should extract file name from LOCALFILE path using pattern', () => {
-    const filePath = '/user/data/:myDocument;v1.csv';
-    const result = getDefaultTableName(filePath, ImporterFileSource.LOCAL);
+  it('should extract file name from LOCAL file metadata with fileName provided', () => {
+    const fileMetaData = {
+      path: '/user/data/myDocument.csv',
+      fileName: 'myDocument.csv',
+      source: ImporterFileSource.LOCAL
+    };
+    const result = getDefaultTableName(fileMetaData);
     expect(result).toBe('myDocument');
   });
 
-  it('should return empty string if LOCALFILE pattern does not match', () => {
-    const filePath = '/user/data/myDocument.csv';
-    const result = getDefaultTableName(filePath, ImporterFileSource.LOCAL);
+  it('should return empty string if LOCAL file has no fileName property', () => {
+    const fileMetaData = {
+      path: '/user/data/myDocument.csv',
+      source: ImporterFileSource.LOCAL
+    };
+    const result = getDefaultTableName(fileMetaData);
     expect(result).toBe('');
   });
 
+  it('should handle LOCAL file with complex filename', () => {
+    const fileMetaData = {
+      path: '/user/data/complex-file_name.with.dots.xlsx',
+      fileName: 'complex-file_name.with.dots.xlsx',
+      source: ImporterFileSource.LOCAL
+    };
+    const result = getDefaultTableName(fileMetaData);
+    expect(result).toBe('complex_file_name_with_dots');
+  });
+
+  it('should handle LOCAL file with special characters in filename', () => {
+    const fileMetaData = {
+      path: '/user/data/file@2023#test.csv',
+      fileName: 'file@2023#test.csv',
+      source: ImporterFileSource.LOCAL
+    };
+    const result = getDefaultTableName(fileMetaData);
+    expect(result).toBe('file_2023_test');
+  });
+
+  it('should handle LOCAL file with spaces in filename', () => {
+    const fileMetaData = {
+      path: '/user/data/my file name.csv',
+      fileName: 'my file name.csv',
+      source: ImporterFileSource.LOCAL
+    };
+    const result = getDefaultTableName(fileMetaData);
+    expect(result).toBe('my_file_name');
+  });
+
   it('should extract file name from REMOTE path as last part of path', () => {
-    const filePath = 'https://demo.gethue.com/hue/test-file.csv';
-    const result = getDefaultTableName(filePath, ImporterFileSource.REMOTE);
-    expect(result).toBe('test-file');
+    const fileMetaData = {
+      path: 'https://demo.gethue.com/hue/test-file.csv',
```

Comment:
nit: I was a bit surprised to see real URLs and first thought thet were actually called. I think it is better to use clearly mocked paths in the tests.